client/landing/gutenboarding/onboarding-block/language/index.tsx

```diff
@@ -38,10 +44,16 @@ const LanguageStep: React.FunctionComponent = () => {
 		)
 	);
 
+	// keep a static reference to the previous step
+	const staticPreviousStep = React.useRef( previousStep );
+
 	const history = useHistory();
+	const makePath = usePath();
 
-	const goBack = () => {
-		history.goBack();
+	const goBack = ( lang = '' ) => {
+		staticPreviousStep.current
+			? history.replace( makePath( Step[ staticPreviousStep.current ], lang ) )
```

Comment:
Ah now that I think about it, since we're treating this as page navigation, the language buttons would all be wrapped in `<a href="">` and _that's_ what would cause the history to be "pushed". But that's all inside the language picker so we can't really change it to use a real link.

I find the fact that the language picker isn't treated like a modal (i.e. has an "X" button instead of a "back" button, and it wouldn't update the browser url) makes it feel a bit clumsy. But that's a bigger change than we should address in this PR. Just treat this as grumbling :)